crates/cheatnet/src/state.rs

```diff
@@ -164,10 +164,8 @@ impl NotEmptyCallStack {
     }
 
     pub fn top(&mut self) -> Rc<RefCell<CallTrace>> {
-        let top_val = self.0.pop().unwrap();
-        let borrowed_ref = top_val.call_trace.clone();
-        self.0.push(top_val);
-        borrowed_ref
+        let top_val = self.0.last().unwrap();
+        top_val.call_trace.clone()
```

Comment:
Not sure if this is correct. I thought we wanted to have a reference to the top call